src/shared/XLinkPrivateFields.c

```diff
@@ -72,7 +72,9 @@ streamDesc_t* getStreamById(void* fd, streamId_t id)
     for (stream = 0; stream < XLINK_MAX_STREAMS; stream++) {
         if (link->availableStreams[stream].id == id) {
             XLink_sem_wait(&link->availableStreams[stream].sem);
-
+            if(link->availableStreams[stream].id == INVALID_STREAM_ID){
```

Comment:
This code is unnecessary. The check for `INVALID_STREAM_ID` is already done on line 68.
1. checks if param id=INVALID_STREAM_ID. If it is (which is an invalid param for this func) then returns NULL.
2. After that line of code, it is therefore guaranteed that id != INVALID_STREAM_ID
3. In the for loop...
4. if (link->availableStreams[stream].id == id) checks the slot in the array id with the param id. Remember that the param id is guaranteed != INVALID_STREAM_ID. Therefore, the logic on this line 73 can never be `true` for a slot in the array that has its id=INVALID_STREAM_ID.

this makes your addition if test lines 75-77 duplications. Line 75 can never be true.